core/peerset/src/lib.rs

```diff
@@ -72,6 +75,11 @@ impl PeersetHandle {
 	pub fn report_peer(&self, peer_id: PeerId, score_diff: i32) {
 		let _ = self.tx.unbounded_send(Action::ReportPeer(peer_id, score_diff));
 	}
+
+	/// Modify a priority group
+	pub fn set_priority_group(&self, group_id: String, peers: HashSet<PeerId>) {
```

Comment:
what about small modifications to the group; for situations where we discover a random peer but later realize it would be well-categorized in a given group. One example would be finding collators through discovery (or incoming connections).

With the current API we'd have to somehow get the current `HashSet`, manipulate it by 1, and then `set_priority_group`. Although at the moment I don't see a way to `get_priority_group`.

Reply:
That's a good point. I assumed that the set will always from the chain.